src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -490,13 +490,12 @@ private IEnumerable<Location> GetDiagnosticLocationsWithinTree(SyntaxTree tree)
             }
         }
 
-        internal Diagnostic WithReportDiagnostic(ReportDiagnostic reportAction)
+        internal Diagnostic WithReportDiagnostic(ReportDiagnostic reportAction, bool reportSuppressedDiagnostics)
```

Comment:
>reportSuppressedDiagnostics [](start = 85, length = 27)

I think this function and parameter can benefit from a good doc comment. For example I would not expect a "With..." method to return null. Also, I think "reportSuppressedDiagnostics" name is somewhat confusing. The function doesn't report anything. Perhaps "retainIfSuppressed" better reflects what this function does. Also, consider renaming the function to "WithReportDiagnostic**OrNull**" or something similar, to draw attention to the unconventional behavior.